src/plugins/crypto/botan_operations.cpp

```diff
@@ -275,7 +275,11 @@ int elektraCryptoBotanDecrypt (KeySet * pluginConfig, Key * k, Key * errorKey, K
 		if (decryptor.remaining () > 0)
 		{
 			// decode the "header" flags
-			decryptor.read (static_cast<byte *> (&flags), sizeof (kdb_octet_t));
+			const size_t flagLength = decryptor.read (static_cast<byte *> (&flags), sizeof (kdb_octet_t));
+			if (flagLength != sizeof (kdb_octet_t))
+			{
+				throw std::length_error ("Failed to restore the original data type of the value.");
```

Comment:
who is catching the exception?

Reply:
it is caught directly in line 305

Reply:
Ohh sorry, was cut off by github. Isn't this a bit of abuse for exceptions as control flow? If it is a case that never happens it would be okay, though.